---
title: "Overview of TransPCO and Autism gene modules"
author: "Niyati Jain"
date: '2022-06-27'
output: html_document
---

**Overview of Trans-PCO**

This approach identifies SNPs that are associated with the expression of a group of genes (e.g., genes with correlated expression, genes identified in the same biological pathway). The method  identifies SNPs that have a "trans-association" with a given gene module. Here, trans-association is defined as a SNP affecting expression of genes found on another chromosome. 

PCA-based omnibus (PCO) is a multivariate test that can be used to study the association between a SNP and a gene module. Briefly, it incorporates multiple PCs to improve the power to detect  associations. Even though, the first PC captures the largest variance in the data, it can have less power to detect significant association signals. 

Specifically, the best test statisitc (i.e., yielding the lowest p value) from six tests (PCMinP, PCFisher, PCLC, WI, Wald and VC) is used as the PCO test statistic. The six tests combine PCs in different ways (linear or non linear), thus accounting for different correlation structures between SNPs and gene module expression. Note three of the tests combine PCs by weighting against the inverse of their eignevalues. In this case, PCs with very small eigenvalues would contribute more greatly to the association test, leading to perhaps inflated p-values. After running test simulations, it was determined that only PCs with eigenvalues of at least 0.1 will be used.

**Identifying Gene Modules**

Gene modules can be identified using weighted gene network analysis ($R^2$). Krishnan et al., used another approach (shared k-nearest-neighbors and the Louvanian community finding algorithm) to identify autism-associated functional modules. 

```{r, include=FALSE}
library(tidyverse)
allgenes <- read_csv("data/allgenes_autism_modules.csv")
highlygenes <- read_csv("data/highly_connected_genes_autism_modules.csv")
```

```{r autism_module_explImage, include = TRUE, echo = TRUE, out.width = "700px",out.height="400px"}
library(knitr)
knitr::include_graphics("assets/autism_module_expl.png", error = FALSE)
```

1. All Genes: How many clusters? How many genes per cluster? 
```{r}

allgenes %>% group_by(cluster) %>% tally()
```

2. Highly connected genes: How many clusters? How many genes per cluster? 
```{r}
highlygenes %>% group_by(cluster) %>% tally()
```

**Workflow of running Trans-PCO**

1. Identify gene modules (e.g., WGNA)
2. Prepare Z-score matrix (rows: SNPs and columns: genes in a module)


