---
title: "UKB GWAS Analysis"
author: "Niyati Jain"
date: '2022-07-27'
output: html_document
---

Note: this can be done using command line (dx toolkit) 

1. Build Cohort 

      + File designated as dataset (app*)
      + DNANexus Cohort Browser 
            - Add filter (based on variable of interest)
            - Add columns for covariates (sex, self reported ancestry)
            - Save cohort
      + Jupyter Notebook (SPARK)
            
2. Phenotype input file QC

      + Jupyter Notebook (SPARK)
      
3. Genetic data QC

      + Bulk/Exome/Pop level OQFE variants PLINK
      + Merge chromosome files (C1-C22, exclude sex chromosomes?)
      + Swiss Army Knife app to access PLINK (maf, HWE, etc.)
      
4. GWAS 

      + Swiss Army Knife app to access REGENIE
          - SNPs that passed QC and phenotype file (same as covariates file) as inputs
          - Specify pheno column and covar column (sex, ancestral PCs)

5. Visualize results

      + LocusZoom app
          - Manhattan plots, zoom into regions to see overlapping genes, QQPlot
